src/core/properties/positionproperty.cpp

```diff
@@ -28,115 +28,240 @@
  *********************************************************************************/
 
 #include <inviwo/core/properties/positionproperty.h>
+#include <inviwo/core/util/exception.h>
+#include <inviwo/core/util/foreach.h>
+#include <inviwo/core/network/networklock.h>
+#include <inviwo/core/datastructures/spatialdata.h>
+#include <inviwo/core/datastructures/coordinatetransformer.h>
 
 namespace inviwo {
 
 const std::string PositionProperty::classIdentifier = "org.inviwo.PositionProperty";
 std::string PositionProperty::getClassIdentifier() const { return classIdentifier; }
 
 PositionProperty::PositionProperty(std::string_view identifier, std::string_view displayName,
-                                   Document help, FloatVec3Property position,
-                                   CameraProperty* camera, InvalidationLevel invalidationLevel,
-                                   PropertySemantics semantics)
-    : CompositeProperty(identifier, displayName, std::move(help), invalidationLevel, semantics)
-    , referenceFrame_("referenceFrame", "Space")
-    , position_(position)
-    , positionWorldSpace_(position_.get())
-    , camera_(camera) {
-    referenceFrame_.addOption("world", "World", static_cast<int>(Space::WORLD));
-    if (camera_) {
-        referenceFrame_.addOption("view", "View", static_cast<int>(Space::VIEW));
-    }
-    referenceFrame_.setSelectedValue(static_cast<int>(Space::WORLD));
-    referenceFrame_.setCurrentStateAsDefault();
+                                   Document help, const vec3& position,
+                                   CoordinateSpace coordinateSpace, CameraProperty* camera,
+                                   PropertySemantics positionSemantics,
+                                   InvalidationLevel invalidationLevel)
+    : CompositeProperty{identifier, displayName, std::move(help), invalidationLevel}
+    , camera_{camera}
+    , previousReferenceSpace_{coordinateSpace}
+    , referenceSpace_{"referenceFrame", "Space",
+                      OptionPropertyState<CoordinateSpace>{
+                          .options = {{"world", "World coordinates", CoordinateSpace::World},
+                                      {"view", "View coordinates", CoordinateSpace::View},
+                                      {"clip", "Clip coordinates", CoordinateSpace::Clip}}}
+                          .set("Reference coordinate space for the position below."_help)
+                          .set([coordinateSpace]() -> size_t {
```

Comment:
Might add a OptionPropertyState::set(const T& newSelectedValue) method to make this a bit easier....

Reply:
Might be tricky for an `OptionPropertInt`, as `set` will clash with `set(size_t index)`

https://github.com/inviwo/inviwo/blob/96f60e567482831f60dd481ffc294f892c01e840/include/inviwo/core/properties/optionproperty.h#L128-L131

Reply:
Might name it setSelectedValue?